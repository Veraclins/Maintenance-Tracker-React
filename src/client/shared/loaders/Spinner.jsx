import React from 'react';
import spinner from '../../img/spinner.svg';

const Spinner = () => (
  <React.Fragment>
    <nav className="loader">
      <img src={spinner} alt="loading........" />
    </nav>
  </React.Fragment>
);

export default Spinner;
