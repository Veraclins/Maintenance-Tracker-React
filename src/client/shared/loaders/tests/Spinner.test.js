import React from 'react';
import { shallow } from 'enzyme';

import Spinner from '../Spinner';

describe('Test the Spinner component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Spinner />);
    expect(wrapper.find('nav')).toHaveLength(1);
    expect(wrapper.find('img')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
