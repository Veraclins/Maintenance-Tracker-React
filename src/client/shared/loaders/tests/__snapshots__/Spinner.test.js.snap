// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Spinner component renders without crashing 1`] = `
<Fragment>
  <nav
    className="loader"
  >
    <img
      alt="loading........"
      src="test-file-stub"
    />
  </nav>
</Fragment>
`;
